In [38]:
import  pandas as pd 
import numpy as np
import matplotlib.pyplot as plt

In [39]:
#reading the dataset
df = pd.read_csv("job_descriptions.csv")

In [40]:
#checking the current number of rows of the dataset
len(df)

215940

## Data preparation
our data previously had 1600000 rows, so we droped the last 1400000 rows and we permently set our original dataset to the new dataset with the 215940 first rows.

In [42]:
#dropping the last 1,400,000 rows 
#jobs_df1 = df.iloc[:-1400000]
#checking the new lenght
#len(jobs_df1)
#saving the modified dataframe to the csv file
#jobs_df1.to_csv("job_descriptions.csv", index=False)
#dropping the columns "latitude" and "longitude"
#jobs_df1 = jobs_df1.drop(columns=['latitude','longitude'])

In [44]:
jobs_df1.head(20)

Job Id     Experience Qualifications Salary Range  \
0   1089843540111562  5 to 15 Years         M.Tech    $59K-$99K   
1    398454096642776  2 to 12 Years            BCA   $56K-$116K   
2    481640072963533  0 to 12 Years            PhD   $61K-$104K   
3    688192671473044  4 to 11 Years            PhD    $65K-$91K   
4    117057806156508  1 to 12 Years            MBA    $64K-$87K   
5    116831420231957  4 to 12 Years            MCA    $59K-$93K   
6   1292168246729889  3 to 15 Years            PhD   $63K-$103K   
7   1498778686197107   2 to 8 Years          M.Com   $65K-$102K   
8   1680293940995740   2 to 9 Years            BBA   $65K-$102K   
9    255627812588102  1 to 10 Years            BBA    $60K-$80K   
10  2696958764033354  3 to 10 Years            BCA   $57K-$104K   
11  1446194141960546  4 to 12 Years         B.Tech    $64K-$98K   
12  1914121205954296  3 to 15 Years            MCA   $65K-$122K   
13   290761760113904   1 to 8 Years          B.Com    $56K-$86K   
14  1627539131873813   1 to 9 Years            MCA    $57K-$98K   
15  2691974960988857  4 to 12 Years          M.Com   $65K-$100K   
16  2823887164112482  5 to 14 Years         M.Tech    $60K-$83K   
17    76902098772934  0 to 11 Years             BA   $55K-$117K   
18   235216486127884  3 to 12 Years             BA   $55K-$121K   
19   407980927519454   5 to 9 Years            PhD   $65K-$128K   

                location                       Country  Work Type  \
0                Douglas                   Isle of Man     Intern   
1               Ashgabat                  Turkmenistan     Intern   
2                  Macao              Macao SAR, China  Temporary   
3             Porto-Novo                         Benin  Full-Time   
4               Santiago                         Chile     Intern   
5               Brussels                       Belgium  Full-Time   
6            George Town                Cayman Islands  Temporary   
7             São Tomé         Sao Tome and Principe   Contract   
8                   Male                      Maldives  Temporary   
9           Saint John's           Antigua and Barbuda  Full-Time   
10                Manama                       Bahrain   Contract   
11  The City of Hamilton                       Bermuda   Contract   
12              Kingston                       Jamaica  Part-Time   
13                Banjul                        Gambia  Temporary   
14              Damascus          Syrian Arab Republic  Full-Time   
15                 Sanaa                         Yemen  Part-Time   
16            San Marino                    San Marino  Part-Time   
17               Papeete              French Polynesia  Full-Time   
18                 Seoul                   North Korea   Contract   
19              Kinshasa  Democratic Republic Of Congo  Temporary   

    Company Size Job Posting Date Preference  ...                Contact  \
0          26801       2022-04-24     Female  ...   001-381-930-7517x737   
1         100340       2022-12-19     Female  ...           461-509-4216   
2          84525       2022-09-14       Male  ...             9687619505   
3         129896       2023-02-25     Female  ...  +1-820-643-5431x47576   
4          53944       2022-10-11     Female  ...      343.975.4702x9340   
5          23196       2023-07-25       Male  ...    (973)791-5355x52199   
6          26119       2023-04-10       Both  ...   001-268-510-4362x789   
7          40558       2022-09-20     Female  ...     667.202.6824x15893   
8         105343       2022-02-19     Female  ...    +1-337-946-9956x550   
9         102069       2022-05-13       Both  ...   001-318-990-0531x978   
10        130338       2023-07-01     Female  ...       001-683-879-1350   
11        117285       2021-10-11       Male  ...       001-388-421-1960   
12         79071       2022-01-17       Both  ...           200-851-9382   
13        127900       2022-05-24     Female  ...     (229)625-8850x6916   
14         92128

In [46]:
#getting the list of columns names
col_names= list(jobs_df1)
print(col_names)

['Job Id', 'Experience', 'Qualifications', 'Salary Range', 'location', 'Country', 'Work Type', 'Company Size', 'Job Posting Date', 'Preference', 'Contact Person', 'Contact', 'Job Title', 'Role', 'Job Portal', 'Job Description', 'Benefits', 'skills', 'Responsibilities', 'Company', 'Company Profile']


In [47]:
#dropping the columns that are not going to be use 
jobs_df1 = jobs_df1.drop(columns=['Contact','Contact Person'])

In [48]:
#preview of the dataset after the removal of the columns
jobs_df1.head(15)

Job Id     Experience Qualifications Salary Range  \
0   1089843540111562  5 to 15 Years         M.Tech    $59K-$99K   
1    398454096642776  2 to 12 Years            BCA   $56K-$116K   
2    481640072963533  0 to 12 Years            PhD   $61K-$104K   
3    688192671473044  4 to 11 Years            PhD    $65K-$91K   
4    117057806156508  1 to 12 Years            MBA    $64K-$87K   
5    116831420231957  4 to 12 Years            MCA    $59K-$93K   
6   1292168246729889  3 to 15 Years            PhD   $63K-$103K   
7   1498778686197107   2 to 8 Years          M.Com   $65K-$102K   
8   1680293940995740   2 to 9 Years            BBA   $65K-$102K   
9    255627812588102  1 to 10 Years            BBA    $60K-$80K   
10  2696958764033354  3 to 10 Years            BCA   $57K-$104K   
11  1446194141960546  4 to 12 Years         B.Tech    $64K-$98K   
12  1914121205954296  3 to 15 Years            MCA   $65K-$122K   
13   290761760113904   1 to 8 Years          B.Com    $56K-$86K   
14  1627539131873813   1 to 9 Years            MCA    $57K-$98K   

                location                Country  Work Type  Company Size  \
0                Douglas            Isle of Man     Intern         26801   
1               Ashgabat           Turkmenistan     Intern        100340   
2                  Macao       Macao SAR, China  Temporary         84525   
3             Porto-Novo                  Benin  Full-Time        129896   
4               Santiago                  Chile     Intern         53944   
5               Brussels                Belgium  Full-Time         23196   
6            George Town         Cayman Islands  Temporary         26119   
7             São Tomé  Sao Tome and Principe   Contract         40558   
8                   Male               Maldives  Temporary        105343   
9           Saint John's    Antigua and Barbuda  Full-Time        102069   
10                Manama                Bahrain   Contract        130338   
11  The City of Hamilton                Bermuda   Contract        117285   
12              Kingston                Jamaica  Part-Time         79071   
13                Banjul                 Gambia  Temporary        127900   
14              Damascus   Syrian Arab Republic  Full-Time         92128   

   Job Posting Date Preference                     Job Title  \
0        2022-04-24     Female  Digital Marketing Specialist   
1        2022-12-19     Female                 Web Developer   
2        2022-09-14       Male            Operations Manager   
3        2023-02-25     Female              Network Engineer   
4        2022-10-11     Female                 Event Manager   
5        2023-07-25       Male               Software Tester   
6        2023-04-10       Both                       Teacher   
7        2022-09-20     Female                UX/UI Designer   
8        2022-02-19     Female                UX/UI Designer   
9        2022-05-13       Both               Wedding Planner   
10       2023-07-01     Female                    QA Analyst   
11       2021-10-11       Male           Litigation Attorney   
12       2022-01-17       Both           Mechanical Engineer   
13       2022-05-24     Female         Network Administrator   
14       2022-03-01       Male               Account Manager   

                              Role           Job Portal  \
0             Social Media Manager             Snagajob   
1           Frontend Web Developer             Idealist   
2          Quality Control Manager         Jobs2Careers   
3        Wireless Network Engineer             FlexJobs   
4               Conference Manager         Jobs2Careers   
5        Quality Assurance Analyst             Snagajob   
6                Classroom Teacher             FlexJobs   
7          User Interface Designer               Indeed   
8             Interaction Designer               Indeed   
9               Wedding Consultant  Stack Overflow Jobs   
10  Performance Testing Specialist            Glas

In [49]:
#getting some information about the dataset 
jobs_df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215940 entries, 0 to 215939
Data columns (total 19 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   Job Id            215940 non-null  int64 
 1   Experience        215940 non-null  object
 2   Qualifications    215940 non-null  object
 3   Salary Range      215940 non-null  object
 4   location          215940 non-null  object
 5   Country           215940 non-null  object
 6   Work Type         215940 non-null  object
 7   Company Size      215940 non-null  int64 
 8   Job Posting Date  215940 non-null  object
 9   Preference        215940 non-null  object
 10  Job Title         215940 non-null  object
 11  Role              215940 non-null  object
 12  Job Portal        215940 non-null  object
 13  Job Description   215940 non-null  object
 14  Benefits          215940 non-null  object
 15  skills            215940 non-null  object
 16  Responsibilities  215940 non-null  obj